### PR TITLE
 Fix connector for Ballerina 0.990.0

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: jira7/jira_connector.bal [hunk 1]
```diff
@@ -19,834 +19,759 @@
 import ballerina/http;
 import ballerina/log;
 
-# Represents Jira Client Connector Object.
-public type JiraConnector object {
+# Object to initialize the connection with Jira.
+#
+# + jiraClient - The HTTP Client
+public type JiraConnector client object {
```
Comment (ldclakmal) [inline jira7/jira_connector.bal]: Shall we remove this `.bal` file after moving the implementation into `jira_endpoint.bal`.
Please refer https://github.com/wso2-ballerina/module-twitter/pull/74

File: jira7/jira_types.bal [hunk 2]
```diff
@@ -336,24 +338,22 @@ public type IssueRequest record {
...
+    string statusId = "";
+    IssueType issueType = {};
 };
 
 # Represent Jira Connector based errors.
 # + type - Type of the error (HTTP error,server error etc.)
 # + message - Error message
 # + jiraServerErrorLog - Error log returned by the jira server, for "server error" type
-# + cause - Cause for the error
 public type JiraConnectorError record {
```
Comment (ldclakmal) [inline jira7/jira_types.bal]: Please remove this record and use ballerina `error`.

File: jira7/jira_data_mappings.bal [hunk 3]
```diff
@@ -22,8 +22,7 @@ import ballerina/mime;
 
 function errorToJiraConnectorError(error source) returns JiraConnectorError {
     JiraConnectorError target = {
-        message: source["message"] ?: "",
-        cause: source["cause"] ?: {}
+        message: <string> source.detail().message
```
Comment (ldclakmal) [inline jira7/jira_data_mappings.bal]:
```suggestion
        message: <string>source.detail().message
```

File: jira7/jira_utils.bal [hunk 4]
```diff
@@ -29,38 +29,32 @@ import ballerina/mime;
...
+        } else {
+            JiraConnectorError e = {
+                ^"type": "Jira Server Error",
+                message: "status" + <string>httpConnectorResponse.statusCode + ":" + httpConnectorResponse.reasonPhrase
+            };
+            //Extracting the error response from the JSON payload of the Jira server response
+            if (httpConnectorResponse.getJsonPayload() is json) {
+                e.jiraServerErrorLog = <json> httpConnectorResponse.getJsonPayload();
+                e.jiraServerErrorLog = null;
+                }
```
Comment (ldclakmal) [inline jira7/jira_utils.bal]: Please format here.

File: jira7/tests/tests.bal [hunk 5]
```diff
@@ -298,11 +305,10 @@ function test_getProjectComponent() {
...
     var output = jiraConnectorEP->getProjectComponent(projectComponent_test.id);
-    match output {
-        ProjectComponent component => {
-            projectComponent_test = untaint component;
-        }
-	JiraConnectorError e => test:assertFail(msg = formatJiraConnError(e));
+    if (output is ProjectComponent) {
+        projectCategory_test = untaint output;
+    } else {
+	    test:assertFail(msg = formatJiraConnError(output));
```
Comment (ldclakmal) [inline jira7/tests/tests.bal]: Please format here.

File: jira7/tests/tests.bal [hunk 6]
```diff
@@ -384,11 +393,10 @@ function test_getProjectCategory() {
...
     var output = jiraConnectorEP->getProjectCategory(projectCategory_test.id);
-    match output {
-        ProjectCategory category => {
-            projectCategory_test = untaint category;
-        }
-	JiraConnectorError e => test:assertFail(msg = formatJiraConnError(e));
+    if (output is ProjectCategory) {
+        projectCategory_test = untaint output;
+    } else {
+	    test:assertFail(msg = formatJiraConnError(output));
```
Comment (ldclakmal) [inline jira7/tests/tests.bal]: Please format here.

File: jira7/tests/tests.bal [hunk 7]
```diff
@@ -473,11 +487,10 @@ function test_getIssue() {
...
     var output = jiraConnectorEP->getIssue(issue_test.key);
-    match output {
-        Issue issue => {
-            issue_test = issue;
-        }
-	JiraConnectorError e => test:assertFail(msg = formatJiraConnError(e));
+    if (output is Issue) {
+        issue_test = output;
+    } else {
+	    test:assertFail(msg = formatJiraConnError(output));
```
Comment (ldclakmal) [inline jira7/tests/tests.bal]: Please format here.

File: jira7/jira_types.bal (hunk 2 above)
Comment (ldclakmal) [reply jira7/jira_types.bal]: Do we need to keep this record for the error?

Comment (Biruntha) [reply jira7/jira_types.bal]: I have created an issue to track this[1]

[1] https://github.com/wso2-ballerina/module-jira/issues/49

**Pull request merged**